### PR TITLE
Better error message for MySQL 8 workaround

### PR DESCRIPTION
**Pull request by BilalReffas**
### Summary of Changes
Joomla 4.0 don't work with MySQL 8 without a workaround.

Please follow these introductions from this [wiki](https://docs.joomla.org/Joomla_and_MySQL_8#Workaround_to_get_Joomla_working_with_MySQL_8) I wrote to make Joomla 4.0 and MySQL 8 working.
Joomla 3.8, 3.9 are working fine as well with this workaround.

I add a link to the error message to help some users who wants to use Joomla with MySQL 8.

Before

bildschirmfoto 2018-07-30 um 09 18 08

After

bildschirmfoto 2018-08-01 um 12 18 31

### Test instructions
Please open your configuration file <tt>sudo nano /etc/my.cnf</tt> (Please note that your file may be under a different directory) and add the following configuration:</translate> 
```sql
[mysqld]
default-authentication-plugin=mysql_native_password
```
<translate> If you don't have access to your config file then you can update your user as follows:</translate>
```sql
ALTER USER 'username'@'localhost' IDENTIFIED WITH mysql_native_password BY 'password';
```
<translate> Replace username with the name of the user account and password with the password belonging to the account. Restart MySQL and you are done, well only if you have Joomla 3.8 or 3.9 installed.</translate>

<translate> If you want to work with Joomla 4.0 you have to do some extra work, because the administration dashboard is blank after the installation with Joomla 4.0 and MySQL 8.</translate>

<translate>
Fortunately there is a fix for that too! 
MySQL 8 is returning <tt>Incorrect date value: '0000-00-00'</tt> after running following insertion from the installation file <tt>/installation/sql/mysql/joomla.sql</tt>.</translate> 

<translate> The following query must be run to fix this. Replace #_ with your table prefix.</translate>
```sql
UPDATE `#__modules` SET `checked_out_time` = '1000-01-01 00:00:00', `publish_up` = '1000-01-01 00:00:00', `publish_down` = '1000-01-01 00:00:00';
```

<translate> This is happening because since MySQL 5.7, MySQL stops supporting zeros value in date / datetime. Restart MySQL and you are done!</translate>

<translate>

### CODE REVIEW

**Comment by brianteeman** [comment]
Thank you for this but wouldnt it be better to make joomla work with mysql8 that point people to a wiki

**Comment by BilalReffas** [comment]
This is explained in the wiki, its not a Joomla issue, it's a [PHP issue](https://bugs.php.net/bug.php?id=76660)

**Comment by brianteeman** [comment]
In that case your documentation is wrong. As I read that link to the php issue then its only for a specific version of php 7.2.8 and not generic to mysql8 and joomla. Your documentation implies that the problem is for everyone trying to use joomla with mysql8

**Comment by roland-d** [comment]
@brianteeman The issue @BilalReffas linked in his last comment is the wrong one as that indeed seems related to a specific PHP version. However in the wiki we have the link to the PHP 7.3 commit, which is this one: https://github.com/php/php-src/commit/d6e81f0bfd0cb90586dd83d4fd47a4302605261a

This fixes the MySQL 8 authorization in PHP.

> Your documentation implies that the problem is for everyone trying to use joomla with mysql8

That is exactly the case. The reason is because they changed the password hashing. This is not something we can fix in Joomla, the PHP MySQL client must be updated.

This is not only Joomla but any PHP application and even Sequel Pro on Mac for example. These won't work until the clients are updated to handle the new password mechanism.

**Comment by brianteeman** [comment]
Surely the correct way to "resolve" this is to add a test for mysql8 in the preinstall check and display any message there. Displaying it here on every database installation error will just confuse people - remember most users wouldnt know if they were using mysql8 or not.

**Comment by mbabker** [comment]
> Surely the correct way to "resolve" this is to add a test for mysql8 in the preinstall check and display any message there.

Except you can't.  Because to get the database version, the client has to be able to connect to the server, and right now we can't even reliably establish a connection.

**Comment by BilalReffas** [comment]
Yeah correct @mbabker

File: installation/language/en-GB/en-GB.ini [hunk 1]
```diff
@@ -64,7 +64,7 @@ INSTL_DATABASE_PASSWORD_DESC="Either a password you created or a password provid
 INSTL_DATABASE_PREFIX_DESC="Enter a table prefix or use the randomly generated one."
 INSTL_DATABASE_TYPE_DESC="Select the database type."
 INSTL_DATABASE_USER_DESC="Either a username you created or a username provided by your host."
-INSTL_DATABASE_VALIDATION_ERROR="Check your database credentials, database type, database name or hostname"
+INSTL_DATABASE_VALIDATION_ERROR="Check your database credentials, database type, database name or hostname. You may have MySQL 8 installed then please use this <a href="https://docs.joomla.org/Joomla_and_MySQL_8#Workaround_to_get_Joomla_working_with_MySQL_8">wiki</a> for more information."
```
Comment (brianteeman) [inline installation/language/en-GB/en-GB.ini]: Please change to
"If you have" and "please read this"

Comment (BilalReffas) [reply installation/language/en-GB/en-GB.ini]: Ok 👍

**Comment by brianteeman** [comment]
This still seems very wrong to me. How are other lamp apps handling this?

**Comment by BilalReffas** [comment]
Other applications are using the `mysql_native_password` as a workaround as well.

**Comment by brianteeman** [comment]
@lefred any thoughts on this?

**Comment by brianteeman** [comment]
@BilalReffas Also your wiki article says
> The following queries must be run to fix this. Replace #_ with your table prefix.

But there is only one query

**Comment by BilalReffas** [comment]
Thank you I will change the wiki @brianteeman. Just for clarification is just one query.

File: installation/language/en-GB/en-GB.ini [hunk 2]
```diff
@@ -64,7 +64,7 @@ INSTL_DATABASE_PASSWORD_DESC="Either a password you created or a password provid
 INSTL_DATABASE_PREFIX_DESC="Enter a table prefix or use the randomly generated one."
 INSTL_DATABASE_TYPE_DESC="Select the database type."
 INSTL_DATABASE_USER_DESC="Either a username you created or a username provided by your host."
-INSTL_DATABASE_VALIDATION_ERROR="Check your database credentials, database type, database name or hostname"
+INSTL_DATABASE_VALIDATION_ERROR="Check your database credentials, database type, database name or hostname. If may have MySQL 8 installed then please read this <a href="https://docs.joomla.org/Joomla_and_MySQL_8#Workaround_to_get_Joomla_working_with_MySQL_8">wiki</a> for more information."
```
Comment (brianteeman) [inline installation/language/en-GB/en-GB.ini]: If you not if may ;)

Comment (BilalReffas) [reply installation/language/en-GB/en-GB.ini]: uppsss....

**Review by brianteeman — APPROVED** [review]
approve the language string

**Comment by BilalReffas** [comment]
Some updates on this?

**Comment by ghost** [comment]
@BilalReffas next Step: PR needs minimum 2 successfully Tests.

**Comment by BilalReffas** [comment]
Who will do it @franz-wohlkoenig  ? :)

**Comment by ghost** [comment]
all you can do is give Test Instructions which Volunteers can understand (please consider not all are Developers, maybe most of them).

**Comment by BilalReffas** [comment]
Everyone should follow these introductions from this [wiki](https://docs.joomla.org/Joomla_and_MySQL_8#Workaround_to_get_Joomla_working_with_MySQL_8) I wrote.

**Comment by ghost** [comment]
I would write the Steps here too to make it as easy as possible for Tester to decide if they can/want test and don't have to look on another Site.

**Comment by BilalReffas** [comment]
Okay I will update the text of my pull request

**Comment by ghost** [comment]
hope this works.

**Comment by BilalReffas** [comment]
I updated the test instructions ✅

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 2dc281d6ca5f9241e1b32575d5ac0e67a60572ce<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21339">issues.joomla.org/tracker/joomla-cms/21339</a>.</sub>

**Comment by bene-we** [comment]
I have tested this item :white_check_mark: successfully on 2dc281d6ca5f9241e1b32575d5ac0e67a60572ce<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21339">issues.joomla.org/tracker/joomla-cms/21339</a>.</sub>

File: installation/language/en-GB/en-GB.ini (hunk 2 above)
Comment (Quy) [reply installation/language/en-GB/en-GB.ini]: @BilalReffas Reminder to fix text.

**Comment by BilalReffas** [comment]
Some updates on this?

**Comment by brianteeman** [comment]
Patience is a virtue

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by ghost** [comment]
@BilalReffas next Step: Maintainer decide if merge.

**Comment by wilsonge** [comment]
Looks good to me! Thanks!

**Comment by BilalReffas** [comment]
Thank you @wilsonge !

**Pull request merged**